src/coreclr/src/pal/src/thread/thread.cpp

```diff
@@ -1479,6 +1491,43 @@ CorUnix::GetThreadTimesInternal(
         pTargetThread->Unlock(pThread);
         goto SetTimesToZero;
     }
+#elif defined(__sun)
+    timestruc_t ts;
+    int readResult;
+    char statusFilename[64];
+    snprintf(statusFilename, sizeof(statusFilename), "/proc/%d/lwp/%d/lwpstatus", getpid(), pTargetThread->GetLwpId());
+    fd = open(statusFilename, O_RDONLY);
+    if (fd == -1)
+    {
+       ASSERT("open(%s) failed; errno is %d (%s)\n", statusFilename, errno, strerror(errno));
+       SetLastError(ERROR_INTERNAL_ERROR);
+       pTargetThread->Unlock(pThread);
+       goto SetTimesToZero;
+    }
+
+    lwpstatus_t status;
+
+    readResult = read(fd, &status, sizeof(status));
```

Comment:
The read could be interrupted (return -1 with errno==EINTR), so we need to have a loop here, something like
```suggestion
    do
    {
        readResult = read(fd, &status, sizeof(status));
    }
    while ((readResult == -1) && (errno == EINTR));
```

Reply:
Thank you. I have also removed the error handling case below.

Reply:
@janvorli, just curious if open or close fail should we retry them too?

Reply:
It seems you are right we should not hit other errors listed in the `read` documentation here, but I'd add an _ASSERTE here verifying that readResult != -1.

Reply:
Sure, previously we had an assert under the condition:

```c
    readResult = read(fd, &status, sizeof(status));
    if (readResult == -1)
    {
       ASSERT("read(%s, &status, sizeof(status)) failed; errno is %d (%s)\n", statusFilename, errno, strerror(errno));
       SetLastError(ERROR_INTERNAL_ERROR);
       pTargetThread->Unlock(pThread);
       goto SetTimesToZero;
    }
```

Should we ust revert to that state?

Reply:
Based on the doc for `open`, it should never fail with EINTR for the /proc file system. As for close, we never check its return code at other places and so I think we should not do it here either.

Reply:
> Should we ust revert to that state?

I would just use the _ASSERTE instead to prevent having dead code in the Release build. Unless you believe the read could actually fail with some other error.

Reply:
Makes sense. I have kept the do-while and removed error handling from close().